include/cuco/detail/hash_functions/murmurhash3.cuh

```diff
@@ -355,4 +355,206 @@ struct MurmurHash3_x64_128 {
   MurmurHash3_fmix64<std::uint64_t> fmix64_;
   std::uint64_t seed_;
 };
+
+/**
+ * @brief A `MurmurHash3_x86_128` hash function to hash the given argument on host and device.
+ *
+ * MurmurHash3_x86_128 implementation from
+ * https://github.com/aappleby/smhasher/blob/master/src/MurmurHash3.cpp
+ * -----------------------------------------------------------------------------
+ * MurmurHash3 was written by Austin Appleby, and is placed in the public domain. The author
+ * hereby disclaims copyright to this source code.
+ *
+ * Note - The x86 and x64 versions do _not_ produce the same results, as the algorithms are
+ * optimized for their respective platforms. You can still compile and run any of them on any
+ * platform, but your performance with the non-native version will be less than optimal.
+ *
+ * @tparam Key The type of the values to hash
+ */
+template <typename Key>
+struct MurmurHash3_x86_128 {
+  using argument_type = Key;  ///< The type of the values taken as argument
+  using result_type = cuda::std::array<std::uint32_t, 4>;  ///< The type of the hash values produced
+
+  /**
+   * @brief Constructs a MurmurHash3_x86_128 hash function with the given `seed`.
+   *
+   * @param seed A custom number to randomize the resulting hash value
+   */
+  __host__ __device__ constexpr MurmurHash3_x86_128(std::uint32_t seed = 0)
+    : fmix32_{0}, seed_{seed}
+  {
+  }
+
+  /**
+   * @brief Returns a hash value for its argument, as a value of type `result_type`.
+   *
+   * @param key The input argument to hash
+   * @return The resulting hash value for `key`
+   */
+  constexpr result_type __host__ __device__ operator()(Key const& key) const noexcept
+  {
+    return compute_hash(reinterpret_cast<std::byte const*>(&key),
+                        cuco::extent<std::size_t, sizeof(Key)>{});
+  }
+
+  /**
+   * @brief Returns a hash value for its argument, as a value of type `result_type`.
+   *
+   * @tparam Extent The extent type
+   *
+   * @param bytes The input argument to hash
+   * @param size The extent of the data in bytes
+   * @return The resulting hash value
+   */
+  template <typename Extent>
+  constexpr result_type __host__ __device__ compute_hash(std::byte const* bytes,
+                                                         Extent size) const noexcept
+  {
+    constexpr std::uint32_t block_size = 16;
+    auto const nblocks                 = size / block_size;
+
+    std::uint32_t h1 = seed_;
+    std::uint32_t h2 = seed_;
+    std::uint32_t h3 = seed_;
+    std::uint32_t h4 = seed_;
+
+    constexpr std::uint32_t c1 = 0x239b961b;
+    constexpr std::uint32_t c2 = 0xab0e9789;
+    constexpr std::uint32_t c3 = 0x38b34ae5;
+    constexpr std::uint32_t c4 = 0xa1e38b93;
+    //----------
+    // body
+    for (std::remove_const_t<decltype(nblocks)> i = 0; size >= block_size && i < nblocks; i++) {
```

Comment:
Does `cuda::std::remove_const_t` exist? Prefer `cuda::std::` APIs over `std::` ones in all host-device functions if available.